src/main/scala/scala/collection/mutable/Bag.scala

```diff
@@ -43,18 +43,18 @@ trait Bag[A]
   def -=(elemCount: (A, Int)): this.type = remove(elemCount._1, elemCount._2)
 
   def remove(elem: A, count: Int): this.type = {
-    val amount = Math.max(this.multiplicity(elem) - count, 0)
+    val amount = Math.min(this.multiplicity(elem), count)
     if (amount > 0)
       this.getBucket(elem) match {
-        case Some(b) => b.remove(elem, amount)
+        case Some(b) => updateBucket(b.remove(elem, amount))
         case None => updateBucket((bagConfiguration.newBuilder(elem) add(elem, count)).result())
```

Comment:
... what about this line of code. If I understand this correctly, this will *add* the element in with the given multiplicity if it wasn't present at all in the bag. That can't be right, surely? Was this tested for?